From d5a73dcf09010f97ebdff197991f77eb9992f09e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=8D=C3=B1igo=20Huguet?= <ihuguet@redhat.com>
Date: Fri, 4 Jun 2021 13:36:33 +0200
Subject: [PATCH] net:cxgb3: fix incorrect work cancellation
MIME-Version: 1.0
Git-commit: d5a73dcf09010f97ebdff197991f77eb9992f09e
Patch-mainline: v5.14-rc1
References: bsn#58
Modified-by-SEL: No

Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In my last changes in commit 5e0b8928927f I introduced a copy-paste bug,
leading to cancel twice qresume_task work for OFLD queue, and never the
one for CTRL queue. This patch cancels correctly both works.

Signed-off-by: Íñigo Huguet <ihuguet@redhat.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/chelsio/cxgb3/sge.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/chelsio/cxgb3/sge.c b/drivers/net/ethernet/chelsio/cxgb3/sge.c
index 115c03b0feb6..cb5c79c43bc9 100644
--- a/drivers/net/ethernet/chelsio/cxgb3/sge.c
+++ b/drivers/net/ethernet/chelsio/cxgb3/sge.c
@@ -3310,7 +3310,7 @@ void t3_sge_stop(struct adapter *adap)
 		struct sge_qset *qs = &adap->sge.qs[i];
 
 		cancel_work_sync(&qs->txq[TXQ_OFLD].qresume_task);
-		cancel_work_sync(&qs->txq[TXQ_OFLD].qresume_task);
+		cancel_work_sync(&qs->txq[TXQ_CTRL].qresume_task);
 	}
 }
 
-- 
2.26.2

